components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/choreo/ChoreoStatisticsPublisher.java

```diff
@@ -0,0 +1,293 @@
+import org.apache.synapse.commons.CorrelationConstants;
+import org.apache.synapse.config.SynapsePropertiesLoader;
+import org.apache.synapse.rest.RESTConstants;
+
+import org.wso2.micro.integrator.analytics.messageflow.data.publisher.publish.StatisticsPublisher;
+import org.wso2.micro.integrator.initializer.ServiceBusInitializer;
+
+public class ChoreoStatisticsPublisher implements StatisticsPublisher {
+    private static ChoreoStatisticsPublisher instance = null;
+    private final Log log = LogFactory.getLog(ChoreoStatisticsPublisher.class);
+
+    //These are the flags to disable analytics for each component
+    private boolean analyticsEnabledForAPI;
+    private boolean analyticsEnabledForSequences;
+    private boolean analyticsEnabledForProxyServices;
+    private boolean analyticsEnabledForEndpoints;
+    private boolean analyticsEnabledForInboundEndpoints;
+
+    private boolean analyticsEnabled;
+
+    public static ChoreoStatisticsPublisher GetInstance() {
+        if (instance == null) {
+            instance = new ChoreoStatisticsPublisher();
+        }
+        return instance;
+    }
+
+    /**
+     * Load Configurations.
+     *
+     * This method is used to load configurations from synapse.properties file.
+     */
+    private void loadConfigurations() {
+        analyticsEnabledForAPI = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.API_ANALYTICS_ENABLED, true);
+        analyticsEnabledForSequences = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.SEQUENCE_ANALYTICS_ENABLED, true);
+        analyticsEnabledForProxyServices = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.PROXY_SERVICE_ANALYTICS_ENABLED, true);
+        analyticsEnabledForEndpoints = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.ENDPOINT_ANALYTICS_ENABLED, true);
+        analyticsEnabledForInboundEndpoints = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.INBOUND_ENDPOINT_ANALYTICS_ENABLED, true);
+        analyticsEnabled = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.ANALYTICS_ENABLED, false);
+    }
+
+    protected ChoreoStatisticsPublisher() {
+        loadConfigurations();
+    }
+
+    /**
+     * Process Method of data publisher.
+     *
+     * In this,
+     * It will be checked if analytics is enabled by configuration.
+     * Then, it will be checked if the artifact's analytics is disabled by configuration.
+     * Then, if artifact is Endpoint or Sequence, it will be checked if the artifact is null.
+     *
+     * @param publishingFlow The publishing flow
+     * @param tenantId       The tenant id
+     */
+    @Override
+    public void process(PublishingFlow publishingFlow, int tenantId) {
+        if (!analyticsEnabled) {
```

Comment:
It seems we aren't setting the value of `analyticsEnabled` anywhere. Am I missing something here?

Reply:
In line 115 inside `loadConfigurations()`, we read the configurations. This will be the value of 
```
[analytics]
enabled=true
```
of deployment.toml